muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/workspace/fieldio/WorkspaceJournalFieldIOHandler.java

```diff
@@ -0,0 +1,73 @@
+package fi.otavanopisto.muikku.plugins.workspace.fieldio;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+
+import fi.otavanopisto.muikku.model.users.UserEntity;
+import fi.otavanopisto.muikku.model.workspace.WorkspaceEntity;
+import fi.otavanopisto.muikku.plugins.workspace.WorkspaceJournalController;
+import fi.otavanopisto.muikku.plugins.workspace.WorkspaceMaterialController;
+import fi.otavanopisto.muikku.plugins.workspace.model.WorkspaceJournalEntry;
+import fi.otavanopisto.muikku.plugins.workspace.model.WorkspaceMaterialField;
+import fi.otavanopisto.muikku.plugins.workspace.model.WorkspaceMaterialReply;
+import fi.otavanopisto.muikku.schooldata.WorkspaceEntityController;
+import fi.otavanopisto.muikku.users.UserEntityController;
+
+public class WorkspaceJournalFieldIOHandler implements WorkspaceFieldIOHandler {
+
+  @Inject
+  private WorkspaceJournalController workspaceJournalController;
+  
+  @Inject
+  private WorkspaceMaterialController workspaceMaterialController;
+  
+  @Inject
+  private WorkspaceEntityController workspaceEntityController;
+  @Inject
+  private UserEntityController userEntityController;
+
+  @Override
+  public void store(WorkspaceMaterialField field, WorkspaceMaterialReply reply, String value) throws WorkspaceFieldIOException {
+    String identifier = String.format("%d-%d", field.getId(), reply.getId()); 
+    
+    WorkspaceJournalEntry journalEntry = workspaceJournalController.findJournalEntryByMaterialFieldReplyIdentifier(identifier);
+
+    if (StringUtils.isNotBlank(value)) {
+      if (journalEntry == null) {
+        UserEntity userEntity = userEntityController.findUserEntityById(reply.getUserEntityId());
+        
+          Long workspaceEntityId = workspaceMaterialController.getWorkspaceEntityId(field.getWorkspaceMaterial());
+          WorkspaceEntity workspaceEntity = workspaceEntityController.findWorkspaceEntityById(workspaceEntityId);
+          
+          if (workspaceEntity != null) { 
+          journalEntry = workspaceJournalController.createJournalEntry(workspaceEntity, userEntity, value, identifier, field.getWorkspaceMaterial().getTitle());
+        }
+      } else {
+        journalEntry = workspaceJournalController.updateJournalEntry(journalEntry, field.getWorkspaceMaterial().getTitle(), value);
+      }
+    } else {
```

Comment:
Is this if..else block needed? Even if the value would be blank, shouldn't it still lead to a journal entry being either created or updated?